penguin/tools/browser_harness_tools.py

```
@@ -0,0 +1,870 @@
+    """Raised when the optional browser-harness backend cannot be used."""
+
+
+def _slug(value: str) -> str:
+    value = value.strip() or "default"
+    return re.sub(r"[^A-Za-z0-9_.-]+", "-", value)[:80] or "default"
+
+
+class BrowserHarnessAdapter:
+    """Small adapter around browser-harness helpers and daemon startup."""
+
+    def __init__(
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)